src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java

```diff
@@ -672,4 +665,61 @@ private Either<CheckoutRedirectConfigurationException, URI> getRedirectUrlForPsp
             );
         }
     }
+
+    private Mono<NpgBuildData> retrieveNpgBuildDataInformation(AuthorizationRequestData authorizationRequestData) {
+        return uniqueIdUtils.generateUniqueId()
+                .flatMap(
+                        orderId -> new JwtTokenUtils()
+                                .generateToken(
+                                        npgNotificationSigningKey,
+                                        npgJwtKeyValidityTime,
+                                        new Claims(
+                                                authorizationRequestData.transactionId(),
+                                                orderId,
+                                                authorizationRequestData.paymentInstrumentId()
+                                        )
+                                ).fold(
+                                        Mono::error,
+                                        notificationToken -> new JwtTokenUtils()
+                                                .generateToken(
+                                                        ecommerceSigningKey,
+                                                        jwtEcommerceValidityTimeInSeconds,
+                                                        new Claims(
+                                                                authorizationRequestData.transactionId(),
+                                                                orderId,
+                                                                authorizationRequestData.paymentInstrumentId()
+                                                        )
+                                                ).fold(
+                                                        Mono::error,
+                                                        outcomeToken -> Mono.just(
+                                                                new NpgBuildData(
+                                                                        orderId,
+                                                                        notificationToken,
+                                                                        outcomeToken
+                                                                )
+                                                        )
+                                                )
+                                )
+
+                );
+    }
+
+    private URI generateNpgOutcomeUrl(
```

Comment:
🚲  this method is used by both npg and redirect payment flows. What about refactoring it to 
```suggestion
    private URI generateOutcomeUrl(
```

Reply:
Fixed with c294728dd00f565352271c99e453544d4a954731